org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java

```diff
@@ -12561,7 +12561,7 @@ public boolean scheduleProblemForContext(Runnable problemComputation) {
  * <li>individual methods should ensure that {@link #referenceContext} is reset to {@code null} afterwards. This may
  * happen in two ways:
  * <ul><li>by calling any of the {@code handle(..)} methods on all paths, or failing that
- * <li>wrap their body in {@code try(this) { ... }}.
+ * <li>wrap their body in {@code try { ... } finally {problemReporter.close();}}.
```

Comment:
should be: `this.close()` (this bullet is not about external clients).

Other than that: :+1:

Reply:
it is used several times outside:
![image](https://github.com/eclipse-jdt/eclipse.jdt.core/assets/51790620/a16863c3-9cbd-41ee-b9a4-10f0fa514589)

Reply:
> it is used several times outside:

I was looking at the state before #1946 at which point there were no outside callers. The idea to introduce close() was so that outside callers would need to worry less, rather than more. By deciding against AutoCloseable the clear design is gone, no reason any more to call this method "close" etc.

I leave it to you to rephrase the javadoc in a non-confusing way, which must clarify, which code exactly is obliged to follow that rule. Hint: "individual methods must" was intended to refer to all public methods of `ProblemReporter` (admittedly not fully explicit), which - as you showed me - is no longer suitable after #1946.